sdk/confidentialledger/azure-confidentialledger/azure/confidentialledger/_patch.py

```diff
@@ -0,0 +1,109 @@
+
+__all__: List[str] = [
+    "ConfidentialLedgerCertificateCredential",
+    "ConfidentialLedgerClient",
+]  # Add all objects you want publicly available to users at this package level
+
+
+def patch_sdk():
+    """Do not remove from this file.
+
+    `patch_sdk` is a last resort escape hatch that allows you to do customizations
+    you can't accomplish using the techniques described in
+    https://aka.ms/azsdk/python/dpcodegen/python/customize
+    """
+
+
+class ConfidentialLedgerCertificateCredential:
+    """A certificate-based credential for the ConfidentialLedgerClient.
+
+    :param certificate_path: Path to the PEM certificate to use for authentication.
+    :type certificate_path: Union[bytes, str, os.PathLike]
+    """
+
+    def __init__(self, certificate_path: Union[bytes, str, os.PathLike]):
+        self._certificate_path = certificate_path
+
+    @property
+    def certificate_path(self) -> Union[bytes, str, os.PathLike]:
+        """The path to the certificate file for this credential."""
+
+        return self._certificate_path
+
+
+class ConfidentialLedgerClient(GeneratedClient):
+    """The ConfidentialLedgerClient writes and retrieves ledger entries against the Confidential
+    Ledger service.
+
+    :ivar confidential_ledger: ConfidentialLedgerOperations operations
+    :vartype confidential_ledger: azure.confidentialledger.operations.ConfidentialLedgerOperations
+    :param endpoint: The Confidential Ledger URL, for example
+     https://contoso.confidentialledger.azure.com.
+    :type endpoint: str
+    :param credential: A credential object for authenticating with the Confidential Ledger.
+    :type credential: Union[
+        ~azure.confidentialledger.ConfidentialLedgerCertificateCredential,
+        ~azure.core.credentials.TokenCredential]
+    :param ledger_certificate_path: The path to the Confidential Ledger's TLS certificate. If this
+        file does not exist yet, the Confidential Ledger's TLS certificate will be fetched and saved
+        to this file.
+    :type ledger_certificate_path: Union[bytes, str, os.PathLike]
+    :keyword api_version: Api Version. Default value is "2022-05-13". Note that overriding this
+     default value may result in unsupported behavior.
+    :paramtype api_version: str
+    """
+
+    def __init__(
+        self,
+        endpoint: str,
+        credential: Union[ConfidentialLedgerCertificateCredential, TokenCredential],
+        *,
+        ledger_certificate_path: Union[bytes, str, os.PathLike],
+        **kwargs: Any,
+    ) -> None:
+        if os.path.isfile(ledger_certificate_path) is False:
+            # We'll need to fetch the TLS certificate.
+            identity_service_client = ConfidentialLedgerCertificateClient(**kwargs)
+
+            # Ledger URIs are of the form https://<ledger id>.confidential-ledger.azure.com.
+            ledger_id = endpoint.replace("https://", "").split(".")[0]
```

Comment:
nit: it might be good to do a try catch here to improve the error message if a user passes in a poorly formed ledger uri. Just a nit, bc the error here could look a little weird if it's not a well-formed uri

Reply:
I think this is fine - `replace` is a noop if the string isn't found, while `split` will always return a list of at least length 1, even if the string is empty or doesn't contain `.`, so no exception is possible anyway (unless `endpoint` isn't a string at all, but I don't think we're worried about that).